File: 백준/Silver/14916. 거스름돈/거스름돈.py
# 14916 거스름돈

# 동전의 개수가 최소가 되도록 돌려주기

n = int(input())

count = 0
while n != 0:
    if n % 5 == 0:
        count += n // 5
        break
    else:
        n -= 2
        count += 1
    if n < 0:
        break

if n < 0:
    print(-1)
else:
    print(count)

File: 백준/Bronze/5585. 거스름돈/거스름돈.js
const path = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(path).toString().trim().split("\n");

function solution(input) {
  const N = Number(input[0]);
  let remainder = 1000 - N;

  const coins = [500, 100, 50, 10, 5, 1];
  let count = 0;

  for (const coin of coins) {
    const coinCount = Math.floor(remainder / coin);
    count += coinCount;
    remainder -= coin * coinCount;
  }

  console.log(count);
}

solution(input);
